lib/src/configuration_wire_protocol.dart

```diff
@@ -0,0 +1,230 @@
+enum FormatEnum { precomputed }
+
+/// Dart representation of the Precomputed server response
+class ObfuscatedPrecomputedConfigurationResponse {
+  /// Format of the configuration, always PRECOMPUTED
+  final FormatEnum format = FormatEnum.precomputed;
+
+  /// Always true, indicating this is an obfuscated configuration
+  final bool obfuscated = true;
+
+  /// When the configuration was created
+  final String createdAt;
+
+  /// Optional environment information
+  final String? environment;
+
+  /// Salt used for hashing md5-encoded strings
+  final String salt;
+
+  /// PrecomputedFlag ships values as string and uses ValueType to cast back on the client.
+  /// Values are obfuscated as strings, so a separate Obfuscated interface is not needed for flags.
+  final Map<MD5String, ObfuscatedPrecomputedFlag> flags;
+
+  /// Map of hashed flag keys to obfuscated precomputed bandits
+  final Map<MD5String, ObfuscatedPrecomputedBandit> bandits;
+
+  ObfuscatedPrecomputedConfigurationResponse({
+    required this.createdAt,
+    this.environment,
+    required this.salt,
+    required this.flags,
+    required this.bandits,
+  });
+
+  factory ObfuscatedPrecomputedConfigurationResponse.fromJson(
+    Map<String, dynamic> json,
+  ) {
+    // Safely convert the flags map to Map<String, dynamic>
+    final flagsMap = json['flags'] as Map;
+    final typedFlagsMap = Map<String, dynamic>.from(flagsMap);
+
+    final response = ObfuscatedPrecomputedConfigurationResponse(
+      createdAt: json['createdAt'] as String,
+      environment: json['environment'] != null
+          ? (json['environment'] is String
+              ? json['environment'] as String
+              : (json['environment'] as Map<String, dynamic>)['name'] as String)
+          : null,
+      salt: json['salt'] as String,
+      flags: typedFlagsMap.map(
+        (key, value) {
+          return MapEntry(
+              key,
+              ObfuscatedPrecomputedFlag.fromJson(
+                  value as Map<String, dynamic>));
+        },
+      ),
+      bandits: json['bandits'] != null
+          ? (json['bandits'] as Map).cast<String, dynamic>().map(
+                (key, value) => MapEntry(
+                    key,
+                    ObfuscatedPrecomputedBandit.fromJson(
+                        value as Map<String, dynamic>)),
+              )
+          : {},
+    );
+
+    return response;
+  }
+}
+
+/// Represents a precomputed flag with a value and type
+class ObfuscatedPrecomputedFlag {
+  /// Optional allocation key
+  final MD5String? allocationKey;
```

Comment:
This is the format for the precomputed configuration from the endpoint? Aren't allocation keys base64 encoded? Same with variation keys? How else would we decode them before sending to the event logger callback?

Reply:
Good catch - `MD5String` and `Base64String` are type aliases on `String` and there wasn't enough enforcement, but they were being decode in `_getPrecomputedAssignment` in `lib/src/precompute_client.dart`. likely worth refactoring for clarity